tokio-rustls/src/client.rs

```diff
@@ -58,7 +61,24 @@ where
     ) -> Poll<io::Result<()>> {
         match self.state {
             #[cfg(feature = "early-data")]
-            TlsState::EarlyData(..) => Poll::Pending,
+            TlsState::EarlyData(..) => {
+                let this = self.get_mut();
+
+                // At this time, we have not really established a tls connection,
```

Comment:
Thanks for adding this, that's helpful! I guess the other bit of information that might be useful to mention here is that in the `EarlyData` state, we are waiting for the `AsyncWrite` impl to transition us to `Stream`, so we ignore readiness events from the read side until the write side says we're done writing the early data. Maybe make that more explicit in the comment?

Reply:
I updated the comment.